zero_bin/rpc/src/native/txn.rs

```diff
@@ -357,3 +398,228 @@ fn prestate_tracing_options(diff_mode: bool) -> GethDebugTracingOptions {
+        .filter(|(_callee_addr, code_usage)| code_usage.is_some())
+        .map(|(callee_addr, code_usage)| {
+            (*callee_addr, code_usage.as_ref().unwrap().get_code_hash())
+        })
+        .collect();
+
+    let to_address: Address = tx
+        .to
+        .unwrap_or_else(|| panic!("No `to`-address for tx: {}.", tx.hash));
+
+    // Guard against transactions to a non-contract address.
+    if callee_addr_to_code_hash.contains_key(&to_address).not() {
+        return Ok(jumpdest_table);
+    }
+    let entrypoint_code_hash: H256 = callee_addr_to_code_hash[&to_address];
+
+    // `None` encodes that previous `entry`` was not a JUMP or JUMPI with true
+    // condition, `Some(jump_target)` encodes we came from a JUMP or JUMPI with
+    // true condition and target `jump_target`.
+    let mut prev_jump = None;
+
+    // Contains the previous op.
+    let mut prev_op = "";
+
+    // Call depth of the previous `entry`.  We initialize to 0 as this compares
+    // smaller to 1.
+    let mut prev_depth = 0;
+    // The next available context. Starts at 1. Never decrements.
+    let mut next_ctx_available = 1;
+    // Immediately use context 1;
+    let mut call_stack = vec![(entrypoint_code_hash, next_ctx_available)];
+    next_ctx_available += 1;
+
+    for (step, entry) in structlog_trace.struct_logs.iter().enumerate() {
+        let op = entry.op.as_str();
+        let curr_depth = entry.depth;
+
+        let exception_occurred = prev_entry_caused_exception(prev_op, prev_depth, curr_depth);
+        if exception_occurred {
+            call_stack.pop().expect("Call stack was empty at POP.");
+        }
+
+        debug_assert!(entry.depth as usize <= next_ctx_available);
+        let (code_hash, ctx) = call_stack.last().unwrap();
+        trace!("TX:   {:?}", tx.hash);
+        trace!("STEP: {:?}", step);
+        trace!("STEPS: {:?}", structlog_trace.struct_logs.len());
+        trace!("PREV OPCODE: {}", prev_op);
+        trace!("OPCODE: {}", entry.op.as_str());
+        trace!("CODE: {:?}", code_hash);
+        trace!("CTX:  {:?}", ctx);
+        trace!("EXCEPTION OCCURED: {:?}", exception_occurred);
+        trace!("PREV_DEPTH:  {:?}", prev_depth);
+        trace!("CURR_DEPTH:  {:?}", curr_depth);
+        trace!("{:#?}\n", entry);
+
+        match op {
+            "CALL" | "CALLCODE" | "DELEGATECALL" | "STATICCALL" => {
+                let callee_address = {
+                    // This is the same stack index (i.e. 2nd) for all four opcodes.  See https://ethervm.io/#F1
+                    let mut evm_stack = entry
+                        .stack
+                        .as_ref()
+                        .expect("No stack found in structLog.")
+                        .iter()
+                        .rev();
+                    let callee_raw = *evm_stack
+                        .nth(1)
+                        .expect("Stack must contain at least two values for a CALL instruction.");
```

Comment:
How are the structured logs looking for failed executions due to stack underflows? We want to ensure this stage goes fine even in the case for example of a CALL on an empty stack. The txn should fail as expected, but this will be caught by the prover, we don't want to crash unexpectedly here.

Reply:
This is a good point.  EVM would go in to an exceptional halting state, but I would detect that before processing next `entry`.